src/network/gun/version.2/hash.ts

```diff
@@ -22,26 +22,30 @@ export const hashPostSalt = memoizePromise(async function(postSalt: string) {
 
     const hash = (await Gun.SEA.work(postSalt, hashPair))!
     return hash.substring(0, N)
-})
+}, undefined)
 
 /**
  * @param key - The key need to be hashed
- * @param stableHash - Set this to true if you're writing new code.
- * Unstable hash may cause problem but we cannot just switch to stable hash
- * because it may breaks current data.
+ * @deprecated
  */
-export const hashCryptoKey = memoizePromise(async function(key: CryptoKey, stableHash: boolean) {
+export const hashCryptoKeyUnstable = memoizePromise(async function(key: CryptoKey) {
     const hashPair = `10198a2f-205f-45a6-9987-3488c80113d0`
     const N = 2
 
-    if (stableHash === true) {
-        const jwk = await CryptoKeyToJsonWebKey(key)
-        if (!jwk.x || !jwk.y) throw new Error('Invalid key')
-        const hash = (await Gun.SEA.work(jwk.x! + jwk.y!, hashPair))!
-        return hash.substring(0, N)
-    } else {
-        const jwk = JSON.stringify(await CryptoKeyToJsonWebKey(key))
-        const hash = (await Gun.SEA.work(jwk, hashPair))!
-        return hash.substring(0, N)
-    }
-})
+    const jwk = JSON.stringify(await CryptoKeyToJsonWebKey(key))
+    const hash = (await Gun.SEA.work(jwk, hashPair))!
+    return hash.substring(0, N)
+}, undefined)
+
+/**
+ * @param key - The key need to be hashed
+ */
+export const hashCryptoKey = memoizePromise(async function(key: CryptoKey) {
+    const hashPair = `10198a2f-205f-45a6-9987-3488c80113d0`
+    const N = 2
+
+    const jwk = await CryptoKeyToJsonWebKey(key)
+    if (!jwk.x || !jwk.y) throw new Error('Invalid key')
+    const hash = (await Gun.SEA.work(jwk.x! + jwk.y!, hashPair))!
```

Comment:
Apologies for being laaaaaate to the new payload party, but we really can just get away with only hashing `x` as long as we are on EC. Let's fix that before it gets released and formalized. And may I ask what's exactly in the `hashPair`?

Reply:
hashPair is a random UUID. You can think it is some type of salt but shared to generate a stable output between clients.

Reply:
`jwk.x` or `jwk.x + jwk.y` is not important, we just need a hash, does the current way looks good to you?

Reply:
No, because by only hashing the points we are throwing away algorithmic information previously provided by hashing the entire thing. This breaks the #191 mandate of being able to stably process a multitude of curves.

Reply:
"Hashing the entire thing" does not mean taking the full hash. It means that we used to hash the entire JWK object, which contained information about the algorithm in `jwk.kty + jwk.alg`. We are not including it anymore, which is a short-sighted view especially given that the current salt has no use.

You are trying to go through this quick by dismissing everything without reading them through.

Reply:
The design here is from the https://github.com/DimensionDev/Maskbook/wiki/Data-structure-on-Gun-version-2 . Maskbook is only using the a part of "feature" of the key, it should be stable, but it's content is not important